source.mak

```diff
@@ -1,77 +1,91 @@
 # Shared macros
 
-HEADERS = \
-	args.h ctags.h debug.h entry.h flags.h general.h get.h htable.h keyword.h \
+PARSER_DIR=parsers
+MAIN_DIR=main
+
+MAIN_HEADERS = args.h ctags.h debug.h entry.h flags.h general.h get.h htable.h keyword.h \
 	main.h options.h parse.h parsers.h pcoproc.h read.h routines.h sort.h \
 	strlist.h trashbox.h vstring.h
 
-SOURCES = \
-	ada.c \
-	args.c \
-	ant.c \
-	asm.c \
-	asp.c \
-	awk.c \
-	basic.c \
-	beta.c \
-	c.c \
-	clojure.c \
-	css.c \
-	cobol.c \
-	dosbatch.c \
-	eiffel.c \
-	entry.c \
-	erlang.c \
-	falcon.c \
-	flags.c \
-	flex.c \
-	fortran.c \
-	get.c \
-	go.c \
-	html.c \
-	htable.c \
-	jscript.c \
-	json.c \
-	keyword.c \
-	lisp.c \
-	lregex.c \
-	lua.c \
-	lxcmd.c \
-	main.c \
-	make.c \
-	matlab.c \
-	objc.c \
-	ocaml.c \
-	options.c \
-	parse.c \
-	pascal.c \
-	pcoproc.c \
-	perl.c \
-	php.c \
-	python.c \
-	read.c \
-	rexx.c \
-	routines.c \
-	ruby.c \
-	rust.c \
-	scheme.c \
-	sh.c \
-	slang.c \
-	sml.c \
-	sort.c \
-	sql.c \
-	strlist.c \
-	tcl.c \
-	tex.c \
-	tg.c \
-	trashbox.c \
-	verilog.c \
-	vhdl.c \
-	vim.c \
-	windres.c \
-	yacc.c \
+PARSER_HEADERS =
+
+HEADERS = \
+	$(addprefix $(MAIN_DIR)/,$(MAIN_HEADERS))     \
+	$(addprefix $(PARSER_DIR)/,$(PARSER_HEADERS))
```

Comment:
`addprefix` is probably a GNU make extension, so if we want to keep `source.mak` portable around Makes, we should probably not use it.  And IMO, adding the prefixes manually on the list is not very problematic, the list is static and it's unlikely anyone ever forgets it (as it wouldn't even build if they forgot adding it)